app/controllers/guest_rooms_controller.rb

```diff
@@ -0,0 +1,6 @@
+class GuestRoomsController < ApplicationController
+  def create
+    @guest_room = GuestRoom.create!(guest_id: params[:id], room_id: params[:room_id])
+    redirect_to "/guests/#{@guest_room.guest_id}"
+  end
```

Comment:
This looks good. I like that you're directly creating the `GuestRoom` object rather than finding the room, finding the guest, and THEN making the association. I'm curious if you need line 3 stored in an instance variable since you don't have a view associated with this create action. Similarly, if you chose to get rid of that instance variable, you could just call `params[:id]` on line 4 when trying to interpolate the guest room.

Reply:
One thing to be careful of with the `!` on line 3 is that your code will throw a gross user-side error (on the browser) if for whatever reason that GuestRoom couldn't be created (maybe they went in an invalid room id or something). Those bangs (`!`) can be super helpful in tests, OR they are helpful when you have exception handlers. If you're interested, take a look at exception handlers to see how you can utilize forced errors in your code.